tinygrad/codegen/uops.py

```diff
@@ -350,7 +350,7 @@ def graph_rewrite(sink:UOp, pm:PatternMatcher) -> UOp:
   replace: Dict[UOp, UOp] = {}
   def __inner_rewrite(n:UOp) -> UOp:
     if n in replace: return replace[n]
-    replace_source = (n.op, n.dtype, tuple(__inner_rewrite(y) for y in n.src), n.arg)
+    replace_source = (n.op, n.dtype, (lambda x: tuple(sorted(x) if n.commutative() else x))(__inner_rewrite(y) for y in n.src), n.arg)
```

Comment:
it might be less sorts and separate the rewrite with sort if we sort in the constructor? so `n.src` here is always sorted if n is commutative, instead of only sort in rewrite

Reply:
seemed like a big change but id  love to do that